universal-login-example/src/components/ApproveConnection.js

```diff
@@ -22,10 +27,26 @@ class ApproveConnection extends Component {
     await sdk.denyRequest(identityAddress, address);
   }
 
+  async onRecoverClick() {
+    this.setState({btnLabel: 'Recovering...'});
+    const {identityService, sdk} = this.props.services;
+    let wallet = await Wallet.fromBrainWallet(this.identityService.identity.name, this.state.backupCode);
+    // TODO: We should probably write into the contract a 'Swap' function which adds a key and removes another key in one transaction.
+    await sdk.addKey(identityService.identity.address, identityService.deviceAddress, wallet.privateKey, DEFAULT_PAYMENT_OPTIONS);
+    await sdk.removeKey(identityService.identity.address, wallet.address, wallet.privateKey, DEFAULT_PAYMENT_OPTIONS);
```

Comment:
Oh I see what you meant by "one time use only". This is actually an interesting idea, didn't consider using the key exclusively to login.

There's an issue however: in some cases, adding and removing a key, might require more than one confirmation for security purposes, meaning that you might be in a situation where you're using a key to recover but might need the lost keys to approve it.

In order to simplify it, I would simply encourage users to throw away the recovery paper and create more recovery keys in the future as an optional step.

Reply:
This would probably require a new screen, similar to the greetings screen, or something else, telling the user that. Something similar in layout to other screens we already have:

<img width="648" alt="screen shot 2018-10-04 at 10 40 26 am" src="https://user-images.githubusercontent.com/112898/46477892-1b46ae80-c7c2-11e8-8d54-24453511b071.png">

Reply:
I will make these changes, but I will say I think having a bunch of keys accumulate on an identity is dangerous for security. That means these sheets of paper are free rides into an account permanently unless the user deletes them.

Reply:
Which is why the user should probably burn the little pieces of paper 😉 
But then, they might be useful as recoveries. I think that you'll be correct long term and that the safer way will be to replace the keys when used the first time, but I'd wait until we have the whole multi-approve process smoothed out